x-pack/plugins/security_solution/public/common/components/hover_actions/utils.test.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { act, renderHook } from '@testing-library/react-hooks';
+import { useTopNPopOver } from './utils';
+
+describe('useTopNPopOver', () => {
+  it('calls setIsPopoverVisible when toggling top N', () => {
+    const setIsPopoverVisible = jest.fn();
+    const {
+      result: {
+        current: { toggleTopN },
+      },
+    } = renderHook(() => useTopNPopOver(setIsPopoverVisible));
+
+    act(() => {
+      toggleTopN();
+    });
+
+    expect(setIsPopoverVisible).toHaveBeenCalled();
+  });
+
+  it('calls sets isShowingTopN to true when toggleTopN is called', () => {
```

Comment:
Do you think Is it also necessary to add a test `calls sets isShowingTopN to false when toggleTopN is called for the second time.`

I am guessing by the name that toggle both opens & closed the popover.

Reply:
I think that it is a valid test scenario. I am adding it.